pkg/restapi/webhook/webhook.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package webhook
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/hyperledger/aries-framework-go/pkg/common/log"
+)
+
+const notificationInterval = 2 * time.Second
+const notificationSendTimeout = 2 * time.Second
+
+var logger = log.New("aries-framework/agentd")
+
+type dispatcher struct {
+	client         *http.Client
+	subscriberURLs []string
+}
+
+// StartWebhookDispatcher will launch the webhook dispatcher in a new goroutine.
+// The dispatcher is responsible for sending notifications to subscribers.
+func StartWebhookDispatcher(subscriberURLs []string) {
+	if subscriberURLs != nil {
+		dispatcher := &dispatcher{
+			client:         &http.Client{},
```

Comment:
Do not use default client `&http.Client{}` at least timeout should be here

P.S function can be simplified
```
func StartWebhookDispatcher(subscriberURLs []string) {
	if subscriberURLs == nil {
		logger.Warnf("No subscriber URLs provided. Webhook dispatcher will not start.")
	}

	dispatcher := &dispatcher{
		client:         &http.Client{},
		subscriberURLs: subscriberURLs,
	}
	go dispatcher.start()
}
```

Reply:
and replace `subscriberURLs != nil` to `len(subscriberURLs) != 0`

Reply:
I did the simplification and replaced subscriberURLs != nil with len(subscriberURLs) != 0

I sent you a message regarding the timeout.